src/CMakeLists.txt

```diff
@@ -166,6 +166,12 @@ if(NOT CMAKE_BUILD_TYPE)
       STRING "Default BUILD_TYPE is Debug, other options are: RelWithDebInfo, Release, and MinSizeRel." FORCE)
 endif()
 
+if(CMAKE_C_COMPILER_ID MATCHES "GNU")
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
```

Comment:
cmake's `CMAKE_CXX_FLAGS_DEBUG_INIT` already contains `-g`, so i am not adding it.